app/views/notifications/all.html.erb

```diff
@@ -0,0 +1,39 @@
+<%= render partial: 'messages/subnav' %>
+
+<% if @notifications.present? %>
+  <ol class="comments comments1">
+    <% @notifications.each do |notification| %>
+      <% case notification.notifiable %>
+      <% when Comment %>
+      <% comment = notification.notifiable %>
+        <li class="comments_subtree">
+          <%= render "comments/comment", comment: comment, show_story: true, is_unread: notification.read_at.nil?, show_tree_lines: false, show_folder_control: false %>
+		  <ol class="comments"></ol>
+        </li>
+      <% when Message %>
+      <% message = notification.notifiable %>
+        <li class="comments_subtree">
+          <%= render "message", message: message, is_unread: notification.read_at.nil? %>
+        </li>
+      <% end %>
+    <% end %>
+  </ol>
+<% else %>
+  <p class="help">No notifications to show.</p>
+<% end %>
+
+<% if action_name == "all" && @has_more %>
```

Comment:
```suggestion
<% if @has_more %>
```

I try not to have templates peeking back up the stack at what's rendering them, and in this case it can just decide based on `@has_more`.